### DISCUSSION TITLE
Can I count endpoint data under the service?

### DISCUSSION THREAD

**Discussion by AllenArbiter** [apache/skywalking#10982] [General]
I need to implement "Form Building Time Statistics," which is handled by a separate class. Currently, I intercept the class and create a logical endpoint. It works fine when using it on the endpoint page, but there is no data available when I try to calculate the endpoint's execution time under the service dimension. Am I not allowed to do this?

I studied “service_cpm_http_other = from(Service.*).filter(tags not contain "http.method:GET").cpm() ”  
Wrote my own oal  
```
service_cpm_http_other_typea = from(Service.*).filter(name == "actionsoft.FormEngine.save").cpm()
service_cpm_http_other_typeB = from(Service.*).filter(endpointName == "actionsoft.FormEngine.save").cpm()
service_cpm_http_other_typeC = from(Service.*).filter(tags == "actionsoft.FormEngine.save").cpm()
service_cpm_http_other_typeD = from(Service.*).filter(tags contain "actionsoft.FormEngine.save").cpm()
```

**Comment by wu-sheng** [comment]
Endpoint metric would not available on service. That is out of scope.

**Comment by AllenArbiter** [comment]
I see. I want to refer to the factory example
```
service_cpm_http_get = from(Service.*).filter(tags contain "http.method:GET").cpm()
```
Base and service dimension statistics my class“FormEngine.save Mean response time” How to do it

**Reply by wu-sheng** [reply]
That is not logic endpoint do. Create a metric on your own, and learn MAL.

**Reply by AllenArbiter** [reply]
I also tried using MAL but how can I do the computational response time?
What I'm thinking now is that I should still intercept this class and then use it whenever it's called
``` 
MeterFactory.gauge() 
```
I push the pointer up, but I don't know how to get the start time if I don't have the start time I can't calculate the response time of the intercepted point

**Reply by wu-sheng** [reply]
This is your use case, I actually don't know what it is. The tech possibilities are as above.

**Reply by AllenArbiter** [reply]
Okay, thank you.

**Reply by AllenArbiter** [reply]
Or the question above is I just want to count the execution time of a class and I've learned MAL but I still don't know how to count the execution time of a class

**Reply by wu-sheng** [reply]
Like I said, that is an endpoint metric, and would not be a service metric.

**Reply by AllenArbiter** [reply]
Yes, I understand what you said.... But our requirement is to display it in service。 I used to want to use OAL to write the endpoint name or tag and then count the endpoint and have it displayed under service

**Reply by AllenArbiter** [reply]
Similarly, is it possible to display the endpoint's time metrics under service?

**Reply by wu-sheng** [reply]
Why do you think your requirement must be supported? Clearly, that is a wrong start to the whole thing.

**Reply by AllenArbiter** [reply]
Yeah, you're right. I was wondering how do I count the execution time of a method without using an endpoint

**Reply by wu-sheng** [reply]
There is no simple answer, this is on your own.
